Source/CombatExtended/CombatExtended/Verbs/Verb_LaunchProjectileCE.cs

```diff
@@ -1291,8 +1291,25 @@ private bool CanHitCellFromCellIgnoringRange(Vector3 shotSource, IntVec3 targetL
                 {
                     targetPos = targetLoc.ToVector3Shifted();
                 }
+
                 Ray shotLine = new Ray(shotSource, (targetPos - shotSource));
 
+                foreach (Pawn pawn in shotSource.ToIntVec3().PawnsNearSegment(targetLoc, caster.Map, 1, behind: false, infront: true))
+                {
+                    if (pawn.Faction == ShooterPawn?.Faction)
```

Comment:
Maybe we should use `if (pawn.Faction != null && !ShooterPawn.HostileTo(pawn))` to avoid friendly fire with ally (wandering wild men/animals will be ignored)